lib/IDE/CodeCompletionResultType.cpp

```diff
@@ -217,13 +220,21 @@ const USRBasedType *USRBasedType::fromType(Type Ty, USRBasedTypeArena &Arena) {
     return ExistingTypeIt->second;
   }
 
+  LLVM_DEBUG(llvm::dbgs() << "enter USRBasedType(" << Ty << ", USR = "
+                          << USR << ")\n";
+             Ty->dump(llvm::dbgs()););
+
   SmallVector<const USRBasedType *, 2> Supertypes;
   ;
   if (auto Nominal = Ty->getAnyNominal()) {
     if (auto *Proto = dyn_cast<ProtocolDecl>(Nominal)) {
       Proto->walkInheritedProtocols([&](ProtocolDecl *inherited) {
         if (Proto != inherited &&
-            !inherited->isSpecificProtocol(KnownProtocolKind::Sendable)) {
+            !inherited->isSpecificProtocol(KnownProtocolKind::Sendable) &&
+            !inherited->getInvertibleProtocolKind()) {
```

Comment:
I’m not sure how the inverted protocols are modeled in the AST but shouldn’t we check if the specific conformance is inverted (ie. has `~`) instead of checking whether the protocol is invertible?

Reply:
We're just skipping these protocols entirely here because we get the wrong USRs for their ProtocolDecls. Once the USR mangling is fixed we can remove this hack.